src/plugins/intel_gpu/src/graph/graph_optimizer/prepare_primitive_fusing.cpp

```diff
@@ -211,17 +211,22 @@ void prepare_primitive_fusing::fuse_reorders(program &p) {
                 node.get_dependencies().size() != 1 || input.can_be_optimized())
                 return;
 
-            // - check if previous node is reorder with 1 user (and if the layouts are the same - remove reorder)
+            // - check if previous node is reorder with 1 user
+            //   and if the layouts are the same OR input is NV12 reorder - remove reorder
             // - do not fuse if current node has mean subtract
             if (input.get_users().size() != 1 || !input.is_type<reorder>() ||
-                input.get_output_layout() != node.get_output_layout() || node.has_mean() ||
-                !node.get_primitive()->subtract_per_feature.empty())
+                (input.get_output_layout() != node.get_output_layout() && !input.as<reorder>().get_primitive()->has_surface_input()) ||
```

Comment:
Could you add unit test for this reorders fusion? Something similar to https://github.com/openvinotoolkit/openvino/blob/master/src/plugins/intel_gpu/tests/passes/prepare_primitive_fusing_test.cpp#L23 to ensure that reorder is properly fused.